src/misc/fs_utils.cpp

```diff
@@ -43,6 +45,25 @@ bool is_directory(const std::string& candidate)
 	return std_fs::is_directory(p, ec);
 }
 
+bool is_hidden_by_host(const std::filesystem::path& pathname)
+{
+	assert(!pathname.empty());
+	const auto filename = pathname.filename().string();
+
+	// Filenames that don't start with dot or are the two directory entries
+	// are not hidden by the host
+	if (filename.find(".") != 0 || filename == "." || filename == "..") {
+		return false;
+	}
+
+	// If the filename starts with a dot and is longer than "8.3" or has any
+	// lower-case characters then it's definitely not a DOS filename and is
+	// meant to be hidden.
+	assert(filename[0] == '.');
```

Comment:
This would certainly take care of the `.DS_Store` files, but it would allow files like `.git` or `.config`, etc.

Instead of this,  hjow about only checking the length and letting 4-char long files through? (so the leading dot, plus the extension, which is valid in MS-DOS)

Reply:
> it would allow files like `.git` or `.config`

They have lower-cases, which won't exist when reading DOS-native files (if they were written in the first place by a DOS app); so they'll be blocked too.

> only checking the length and letting 4-char long files through

I'd like to use all three if we can: if a lower-case in there, then it's excluded too (so `.git` would be excluded, but `.GIT` would be OK).

Reply:
Windows and macOS use case-preserving filesystems, I guess that's why I don't like to throw case-sensitiveness into the mix. But ultimately it's of no consequence because we're really dealing with super rare edges cases here (DOS file with a zero byte for filename and some extension).

So let's go with all three then 😄